File: Shopping/ShoppingCommand/src/main/java/nl/avans/domain/models/events/product/StockRemovedEvent.java
package nl.avans.domain.models.events.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import nl.avans.domain.models.events.Event;

import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class StockRemovedEvent implements Event {
    @JsonProperty
    private UUID productId;

    @JsonProperty
    private int amount;
}

File: InventoryCommands/Domain/Events/StockRemovedEvent.cs
using System;

namespace Domain.Events
{
	public class StockRemovedEvent : IEvent
	{
		public Guid EventId { get; set; } = Guid.NewGuid();
		public Guid OrderId { get; set; }

		public Guid ProductId { get; set; }
		public int Amount { get; set; }

		public string EventName => "StockRemoved";
	}
}
